lib/std/mem.zig

```diff
@@ -876,6 +874,63 @@ pub fn indexOfPos(comptime T: type, haystack: []const T, start_index: usize, nee
     return null;
 }
 
+fn boyerMooreHorspoolPreprocess(pattern: []const u8, table: []usize) void {
+    for (table) |*c| {
+        c.* = pattern.len;
+    }
+
+    var i: usize = 0;
+    while (i < pattern.len - 1) : (i += 1) {
```

Comment:
A bit nicer using a `for` loop:

```zig
for (pattern[0..pattern.len-1]) |ch,i| {
    table[ch] = pattern.len - 1 - i;
}
```

Reply:
thanks, didn't know about that syntax. I had to add reverse of this function as well so im keeping this loop for consistency with the reverse function